pkg/webhook/webhook.go

```diff
@@ -120,10 +119,18 @@ func (o *Options) Run() error {
 		return errors.Wrapf(err, "failed to create ScmClient")
 	}
 
-	o.localGitDir, err = ioutil.TempDir("", "git")
+	gitClient, err := git.NewClient(o.gitServerURL, o.gitKind())
 	if err != nil {
-		return errors.Wrapf(err, "failed to create git temp directory")
+		logrus.WithError(err).Fatal("Error getting git client.")
 	}
+	defer func() {
+		err := gitClient.Clean()
```

Comment:
I guess I am a bit surprised to see a `Clean` here. What does a git client have to clean up?
Also, this is a controller, right and we are in the `Run` method, so this defer will execute when the whole controller dies. So is this just a cleanup to do things "right" or does this really try to fix anything and if so what?

Reply:
Lighthouse is really too much of a black box for me atm. I will get a checkout and start poking around a bit. Then I will be able to give better feedback.

Reply:
So fwiw, the "git client" here needs to clean up because it mainly just does clones of repos to check their `OWNERS` files, so in theory when the process dies it should nuke the temp dir. The problem that this change is fixing is that we were doing that wayyyyyyy too much, in fact. So yeah, here, now that I've made it more like Prow behavior (just initing the git client once, not initing it each time we process a webhook), the cleanup should just be "doing things right".